src/components/NewCommentForm.tsx

```diff
@@ -2,7 +1,88 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { CommentData } from '../types/Comment';
+
+interface Props {
+  addNewComment: (newComment: CommentData) => Promise<void>;
+  isSubmitting: boolean;
+}
+
+const initialNewComment = {
+  name: '',
+  email: '',
+  body: '',
+};
+
+const initialErrors = {
+  nameError: '',
+  emailError: '',
+  bodyError: '',
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  addNewComment,
+  isSubmitting,
+}) => {
+  const [commentData, setCommentData] = useState(initialNewComment);
+  const [hasErrorMessage, setHasErrorMessage] = useState(initialErrors);
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setHasErrorMessage(initialErrors);
+
+    const normalizedData = {
+      name: commentData.name.trim(),
+      email: commentData.email.trim(),
+      body: commentData.body.trim(),
+    };
+
+    if (normalizedData.name.length === 0) {
+      setHasErrorMessage(errors => ({
+        ...errors,
+        nameError: 'Name is required',
+      }));
+    }
+
+    if (normalizedData.email.length === 0) {
+      setHasErrorMessage(errors => ({
+        ...errors,
+        emailError: 'Email is required',
+      }));
+    }
+
+    if (normalizedData.body.length === 0) {
+      setHasErrorMessage(errors => ({
+        ...errors,
+        bodyError: 'Enter some text',
+      }));
+    }
+
+    if (
+      normalizedData.name.length === 0 ||
+      normalizedData.email.length === 0 ||
+      normalizedData.body.length === 0
+    ) {
+      return;
+    }
+
+    addNewComment(normalizedData)
+      .then(() => {
+        commentData.body = '';
```

Comment:
Directly modifying the state object like this can lead to unexpected behavior. Use the setCommentData function to update the state instead.